src/AutoMapper/PropertyMap.cs

```diff
@@ -79,20 +80,11 @@ public MemberInfo SourceMember
             }
         }
 
-        public Type SourceType
-        {
-            get
-            {
-                if (CustomExpression != null)
-                    return CustomExpression.ReturnType;
-                if (CustomResolver != null)
-                    return CustomResolver.ReturnType;
-                if(ValueResolverConfig != null)
-                    return typeof(object);
-                return SourceMember?.GetMemberType();
-            }
-        }
-
+        public Type SourceType => CustomExpression?.ReturnType

```

Comment:
These should be ordered by priority I guess.

Reply:
Modified to match `TypeMapPlanBuilder`